crypto/evp/scrypt.c

```diff
@@ -139,7 +139,7 @@ static void scryptROMix(unsigned char *B, uint64_t r, uint64_t N,
         *pV = *pB++;
         *pV |= *pB++ << 8;
         *pV |= *pB++ << 16;
-        *pV |= *pB++ << 24;
+        *pV |= (uint32_t)*pB++ << 24;
```

Comment:
The error from ubsan is:

```
scrypt.c:142:22: runtime error: left shift of 164 by 24 places cannot be represented in type 'int'
```

AFAIU, the left operand is promoted to `int` but then the result of the shift overflows which, according to ISO C99 §6.5.7, is undefined.

In this particular case it doesn't seem to affect the result ([gcc may not treat this as undefined](https://gcc.gnu.org/onlinedocs/gcc/Integers-implementation.html)), and it may be that the other shifts need the cast as well, but I only applied the minimal fix to make ubsan happy.